### DISCUSSION TITLE
Ethernet link up error while sending the DHCP request

### DISCUSSION THREAD

**Discussion by premjithav** [polarfire-soc/polarfire-soc-documentation#289] [Linux build systems for PolarFire SoC]
We are trying to establish Ethernet Connection with the PolarFire SoC FPGA Icicle Kit, but facing some issues. We are using buildroot Linux and booting the linux from the eMMC.

We also followed the steps which are given in your documents but unable to connect to the network. Ethernet with static IP is working fine but while doing the DHCP request, displaying error messages with Link Up/Link Down. 
![Screenshot from 2023-02-06 14-01-37](https://user-images.githubusercontent.com/2322147/216926282-04c2fa7f-564c-4936-950e-e3fed86b9d89.png)

Please give us a suggestion for the same.

**Comment by vfalanis** [comment]
HI @premjithav ,

Could you please let us know which Ethernet connector you are using? Is it J1 connector (next to the MikroBus socket) or J2?

You should be able to get a dynamic IP address when connecting the Icicle Kit to a router using J1. This should be the eth0 interface.

![eth0-buildroot](https://user-images.githubusercontent.com/5502469/218139397-19b86bf9-e8cc-40d4-af78-03ae16b30d7d.png)

If using J2 connector (eth1), you could try the following steps:

```
# ifconfig eth1 up
# udhcpc -i eth1
```
![eth1-buildroot](https://user-images.githubusercontent.com/5502469/218139677-0a482730-4ac5-44fd-9eaa-fcf5f21f3b77.png)

**Comment by premjithav** [comment]
The issue is with the port, the ethernet is only connecting to 1G network. Not working with 10/100.

**Comment by vfalanis** [comment]
Hi @premjithav ,

The usage of both Ethernet ports is now supported in buildroot external since 2023.02 release. I would suggest to try the release and see if that resolves the issues you are facing. Feel free to re-open the discussion if you need any further clarification :)

Thanks
